kse/src/org/kse/gui/dialogs/DViewDHParameters.java

```diff
@@ -0,0 +1,247 @@
+		throws CryptoException {
+
+		super(parent, title, Dialog.ModalityType.DOCUMENT_MODAL);
+		this.dhParameters = dhParams;
+		initComponents(dhParameters);
+	}
+	/**
+	 * Initializes the dialogue panel and associated elements 
+	 * 
+	 * @param dhParams The byte array for the DH Parameters
+	 * @throws CryptoException
+	 */
+	private void initComponents(byte[] dhParams) throws CryptoException {
+		//TODO Generate DH Parameters icon
+	    PemInfo pemInfo = new PemInfo(EB, null, dhParams);
+		jtAreaPem = new JTextArea(PemUtil.encode(pemInfo));
+		jtAreaPem.setToolTipText(res.getString("DViewDHParameters.jtAreaPem.tooltip"));
+		jtAreaPem.setEditable(false);
+		jtAreaPem.setFont(new Font(Font.MONOSPACED, Font.PLAIN, LnfUtil.getDefaultFontSize()));
+		// keep uneditable color same as editable
+		jtAreaPem.putClientProperty("JTextArea.infoBackground", Boolean.TRUE);	
+
+		jbExport = new JButton(res.getString("DViewDHParameters.jbExport.export.text"));
+		jbExport.setToolTipText(res.getString("DViewDHParameters.jbExport.export.tooltip"));
+		PlatformUtil.setMnemonic(jbExport, res.getString("DViewDHParameters.jbExport.mnemonic").charAt(0));
+		
+		jbCopy = new JButton(res.getString("DViewDHParameters.jbCopy.text"));
+		jbCopy.setToolTipText(res.getString("DViewDHParameters.jbCopy.tooltip"));
+		PlatformUtil.setMnemonic(jbCopy, res.getString("DViewDHParameters.jbCopy.mnemonic").charAt(0));
+
+		jbOK = new JButton(res.getString("DViewDHParameters.jbOK.text"));
+		
+		Container pane = getContentPane();
+		pane.setLayout(new MigLayout("insets 10"));
+		pane.add(jtAreaPem, "span");
+		pane.add(jbCopy, "tag Copy");
+		pane.add(jbExport, "tag Export");
+		pane.add(jbOK, "tag Ok");
+	
+		jbOK.addActionListener(evt -> okPressed());
+
+		jbExport.addActionListener(evt -> {
+			try {
+				CursorUtil.setCursorBusy(DViewDHParameters.this);
+				exportPressed();
+			} finally {
+				CursorUtil.setCursorFree(DViewDHParameters.this);
+			}
+		});
+		
+		jbCopy.addActionListener(evt -> {
+			try {
+				CursorUtil.setCursorBusy(DViewDHParameters.this);
+				copyPressed();
+			} finally {
+				CursorUtil.setCursorFree(DViewDHParameters.this);
+			}
+		});
+		
+		setResizable(false);
+
+		getRootPane().setDefaultButton(jbCopy);
+
+		pack();
+
+		SwingUtilities.invokeLater(() -> jbOK.requestFocus());
+    }
+	
+	/**
+	 * Writes Base64 encoded PEM format to a specified file
+	 * <p>
+	 * See RFC 1421 for further information on PEM formatting
+	 * <p>
+	 * @param filePath Accepts a string file path
+	 * @param DEREncodedDHParams Accepts a byte array
+	 */
+    private void writeDHParams(String filePath, byte[] DEREncodedDHParams) {
+        PemWriter pemWrt;
+
+        try {
+            pemWrt = new PemWriter(new FileWriter(filePath));
+            pemWrt.writeObject(new PemObject(EB, DEREncodedDHParams));
+            pemWrt.flush();
+            pemWrt.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+    /**
```

Comment:
Unlike the rest of the file these lines are indented with spaces.

Reply:
Updated. I had to enable the show whitespace characters and found quite a lot of spaces. Everything should now be tabs with spaces removed.